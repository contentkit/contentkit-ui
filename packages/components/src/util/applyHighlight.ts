import { Modifier, EditorState } from 'draft-js'

function applyHighlight (editorState, selection) {
  const contentState = Modifier.applyInlineStyle(
    editorState.getCurrentContent(),
    selection,
    'HIGHLIGHT'
  )
  return EditorState.push(
    editorState,
    contentState,
    'change-inline-style'
  )
}

export default applyHighlight
