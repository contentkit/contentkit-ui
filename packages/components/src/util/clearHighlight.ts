
import { Modifier, EditorState, SelectionState } from 'draft-js'

function clearHighlight (editorState, selection) {
  const contentState = Modifier.removeInlineStyle(editorState.getCurrentContent(), selection, 'HIGHLIGHT')
  const nextEditorState = EditorState.push(editorState, contentState, 'change-inline-style')
  return nextEditorState
  // return EditorState.forceSelection(nextEditorState, selection)
}

export default clearHighlight