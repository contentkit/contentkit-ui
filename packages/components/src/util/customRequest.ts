import { CustomRequestParams } from '../types'

async function customRequest ({ headers, file, action, onSuccess }: CustomRequestParams): Promise<void> {
  const formData = new window.FormData()
  for (let field in action.fields) {
    formData.append(field, action.fields[field])
  }
  formData.append('file', file)
  const resp = await fetch(action.url, {
    method: 'POST',
    body: formData,
    headers: headers
  })
  onSuccess(resp)
}

export default customRequest
