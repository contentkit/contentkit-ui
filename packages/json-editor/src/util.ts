
import { OPEN_SCOPE_DELIMETERS, CHARACTER_PAIRS } from './fixtures'

export const pad = (offset) => new Array(offset).fill(' ').join('')

export const DEFAULT_INDENT = pad(2)


export function isCharacterPair (left: string, right: string) {
  return OPEN_SCOPE_DELIMETERS.includes(left) && right === CHARACTER_PAIRS[left]
}

