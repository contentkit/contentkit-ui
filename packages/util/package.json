{
  "name": "@contentkit/util",
  "version": "0.0.62",
  "private": false,
  "description": "Util",
  "main": "lib/index.js",
  "keywords": [],
  "scripts": {
    "clean": "rimraf lib",
    "lint": "standard \"src/**/*.js\" --fix | snazzy",
    "build": "yarn run clean && tsc -p ./config",
    "watch": "tsc-watch -p ./config",
    "test": "jest"
  },
  "license": "MIT",
  "dependencies": {
    "left-indent": "^0.0.6"
  },
  "peerDependencies": {
    "draft-js": "0.11.6",
    "immutable": "3.7.6"
  },
  "standard": {
    "parser": "babel-eslint",
    "ignore": [
      "node_modules",
      "dist",
      ".cache"
    ]
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org/",
    "access": "public"
  },
  "devDependencies": {
    "babel-jest": "^24.9.0",
    "draft-js": "^0.11.6",
    "immutable": "3.7.6",
    "jest": "^24.9.0",
    "tsc-watch": "3.0.2",
    "typescript": "^3.7.4"
  },
  "jest": {
    "testMatch": [
      "**/src/**/*.test.js"
    ]
  },
  "gitHead": "af83b1d1596f29fc61c474034b39297155903e59"
}
