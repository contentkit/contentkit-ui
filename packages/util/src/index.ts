export * from './transactions'
export * from './fixtures'
export * from './utils'
