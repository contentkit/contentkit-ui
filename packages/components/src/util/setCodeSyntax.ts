import { EditorState, SelectionState, Modifier, EditorChangeType } from 'draft-js'
import { Map } from 'immutable'
import { getContinuousBlocks, ChangeType, createEntity, Entity, Syntax } from '@contentkit/util'

type Store = {
  getEditorState: () => EditorState,
  setEditorState: (editorState: EditorState) => void
}

function setCodeSyntax (value: string, selectionState, { getEditorState, setEditorState }: Store): void {
  const editorState = getEditorState()
  const nextEditorState = createEntity(editorState, selectionState, Entity.CODE, { [Syntax.KEY]: value })
  setEditorState(nextEditorState)
}

export default setCodeSyntax
