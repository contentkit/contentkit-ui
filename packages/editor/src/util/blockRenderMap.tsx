import React from 'react'
import { Map } from 'immutable'
import { Block } from '@contentkit/util'
import { DefaultDraftBlockRenderMap } from 'draft-js'

const CODE_WRAP = <pre className='cke-code-container' />
const OL_WRAP = <ol className='public-DraftStyleDefault-ul' />
const UL_WRAP = <ul className='public-DraftStyleDefault-ul' />

const customBlockRenderMap = Map({

  [Block.UNSTYLED]: {
    element: 'p'
  },
  [Block.BLOCKQUOTE]: {
    element: 'div',
    wrapper: <blockquote />
  },
  [Block.TODO]: {
    element: 'div'
  },
  [Block.IMAGE]: {
    element: 'figure'
  },
  [Block.VIDEO]: {
    element: 'figure'
  },
  [Block.CODE]: {
    element: 'div',
    wrapper: CODE_WRAP
  },
  [Block.OL]: {
    element: 'li',
    wrapper: OL_WRAP
  },
  [Block.UL]: {
    element: 'li',
    wrapper: UL_WRAP
  },
  [Block.TABLE]: {
    element: 'div'
  },
  [Block.GIST]: {
    element: 'div'
  },
  [Block.SCRIPT]: {
    element: 'div'
  }
})

export default DefaultDraftBlockRenderMap.merge(customBlockRenderMap)
