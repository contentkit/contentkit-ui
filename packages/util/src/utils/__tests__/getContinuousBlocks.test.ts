import { getContinuousBlocks, Block } from '@contentkit/util'
import { EditorState, SelectionState, ContentState, ContentBlock, CharacterMetadata } from 'draft-js'
import { List, Repeat } from 'immutable'

const getEditorState = (blockArray, focusKey: string) => {
  const editorState = EditorState.createWithContent(
    ContentState.createFromBlockArray(blockArray)
  )
  return EditorState.acceptSelection(editorState,
    new SelectionState({
      anchorKey: focusKey,
      anchorOffset: 0,
      focusKey: focusKey,
      focusOffset: 0,
      isBackward: false
    })
  )
}

it('getContinuousBlocks 1', () => {
  const blocksArray = [
    new ContentBlock({ key: 'A' }),
    new ContentBlock({
      key: 'B',
      type: Block.CODE,
      text: 'foo',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({
      key: 'C',
      type: Block.CODE,
      text: 'bar',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({
      key: 'D',
      type: Block.CODE,
      text: 'baz',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({ key: 'E' })
  ]

  const editorState = getEditorState(blocksArray, 'C')
  const blocks = getContinuousBlocks(editorState, 'C').toArray()

  expect(blocks).toMatchSnapshot()
})

it('getContinuousBlocks 2', () => {
  const blocksArray = [
    new ContentBlock({ key: 'A' }),
    new ContentBlock({
      key: 'B',
      type: Block.CODE,
      text: 'foo',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({
      key: 'C',
      type: Block.CODE,
      text: 'bar',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({
      key: 'D',
      type: Block.CODE,
      text: 'baz',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({ key: 'E' })
  ]

  const editorState = getEditorState(blocksArray, 'B')
  const blocks = getContinuousBlocks(editorState, 'B').toArray()

  expect(blocks).toMatchSnapshot()
})
