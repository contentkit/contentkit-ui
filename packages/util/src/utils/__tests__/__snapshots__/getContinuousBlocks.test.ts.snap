// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getContinuousBlocks 1 1`] = `
Array [
  Immutable.Record {
    "key": "B",
    "type": "code-block",
    "text": "foo",
    "characterList": Immutable.List [
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
    ],
    "depth": 0,
    "data": Immutable.Map {},
  },
  Immutable.Record {
    "key": "C",
    "type": "code-block",
    "text": "bar",
    "characterList": Immutable.List [
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
    ],
    "depth": 0,
    "data": Immutable.Map {},
  },
  Immutable.Record {
    "key": "D",
    "type": "code-block",
    "text": "baz",
    "characterList": Immutable.List [
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
    ],
    "depth": 0,
    "data": Immutable.Map {},
  },
]
`;

exports[`getContinuousBlocks 2 1`] = `
Array [
  Immutable.Record {
    "key": "B",
    "type": "code-block",
    "text": "foo",
    "characterList": Immutable.List [
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
    ],
    "depth": 0,
    "data": Immutable.Map {},
  },
  Immutable.Record {
    "key": "C",
    "type": "code-block",
    "text": "bar",
    "characterList": Immutable.List [
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
    ],
    "depth": 0,
    "data": Immutable.Map {},
  },
  Immutable.Record {
    "key": "D",
    "type": "code-block",
    "text": "baz",
    "characterList": Immutable.List [
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
    ],
    "depth": 0,
    "data": Immutable.Map {},
  },
]
`;
